packages/relay/test/login.test.ts

```diff
@@ -0,0 +1,398 @@
+import * as crypto from 'crypto'
+import chai from 'chai'
+import chaiHttp from 'chai-http'
+import { expect } from 'chai'
+import * as chaiAsPromise from 'chai-as-promised'
+import nock from 'nock'
+import { deployContracts, startServer } from './environment'
+import { userService } from '../src/services/UserService'
+import { TWITTER_CLIENT_ID, TWITTER_CLIENT_KEY } from '../src/config'
+import { UserRegisterStatus } from '../src/enums/userRegisterStatus'
+import { HTTP_SERVER, CLIENT_URL } from './configs'
+import { ethers } from 'hardhat'
+import { Identity } from '@semaphore-protocol/identity'
+import { UserState } from '@unirep/core'
+import { Server } from 'http'
+
+chai.use(chaiHttp)
+
+const TWITTER_API = 'https://api.twitter.com'
+const mockState = 'state'
+const mockCode = 'testCode'
+const mockCode2 = 'testCode2'
+const mockUserId = '123456'
+const mockUserId2 = '654321'
+const wrongCode = 'wrong-code'
+const token = btoa(`${TWITTER_CLIENT_ID}:${TWITTER_CLIENT_KEY}`)
+
+describe('LOGIN /login', () => {
+    let snapshot: any
```

Comment:
Is snapshot necessary for this test? It seems that some tests are rely on others. You will revert to the snapshot to have a clean storage before executing the next test. If so, you should move revert statement to ```afterEach```

Reply:
You're right, I comment the snapshot and the error occurred on post test. I will resolve this conversation and approve the review.